examples/text_summarization/extractive_summarization_cnndm_aml_distributed.ipynb

```diff
@@ -0,0 +1,795 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Upload the downloaded dataset to AML workspace"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## folder in the workspace where the data is uploaded to\n",
+    "TARGET_DATA_FOLDER = '/bertsum_processed_data'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds = ws.get_default_datastore()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds.upload(src_dir=LOCAL_DATA_FOLDER, target_path=TARGET_DATA_FOLDER)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prepare the local project folder which is mirror to the workspace for the experiment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## local folder to store all the related files to be copied to the workspace\n",
+    "PROJECT_FOLDER = './azureml_exp'\n",
+    "## conda environment name, the yaml file will be copied to the workspace\n",
+    "CONDA_ENV_NAME = \"nlp_gpu\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ENTRY_SCRIPT = \"extractive_summarization_cnndm_distributed_train.py\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Generated conda file: nlp_gpu.yaml\r\n",
+      "\r\n",
+      "To create the conda environment:\r\n",
+      "$ conda env create -f nlp_gpu.yaml\r\n",
+      "\r\n",
+      "To update the conda environment:\r\n",
+      "$ conda env update -f nlp_gpu.yaml\r\n",
+      "\r\n",
+      "To register the conda environment in Jupyter:\r\n",
+      "$ conda activate nlp_gpu\r\n",
+      "$ python -m ipykernel install --user --name nlp_gpu --display-name \"Python (nlp_gpu)\"\r\n",
+      "\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "!mkdir -p {PROJECT_FOLDER}\n",
+    "!python ../../tools/generate_conda_file.py --gpu --name {CONDA_ENV_NAME}\n",
+    "!cp ./nlp_gpu.yaml {PROJECT_FOLDER}\n",
+    "!cp {ENTRY_SCRIPT} {PROJECT_FOLDER}\n",
+    "!cp -r ../../utils_nlp {PROJECT_FOLDER}\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Submit Run"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#AZUREML_CONFIG_PATH = \"./.azureml\"\n",
+    "## output dir in the workspace\n",
+    "MODEL_NAME = \"distilbert-base-uncased\"\n",
+    "ENCODER = \"transformer\"\n",
+    "TARGET_OUTPUT_DIR = f'output/{EXPERIMENT_NAME}/'\n",
```

Comment:
Can we group all imports and constants and place them at the top of the notebook

Reply:
I had all the constants together but decided to not group them together. In this way, I am clear of what constants and import are needed in what step. Is there a style requirement for the repo. Personally I prefer this way. But if we see strong need of grouping them together, I can change the notebook.

Reply:
I think there are two advantages of grouping the constants
- It's clear from one cell what all the constant values are, so it's less likely for someone to forget to change some constants or assume the wrong constant values.
- It will be easier to control all the constants using one one parameter cell in the testing code.

Reply:
The disadvantage is that it's not clear of what constant is being used at what step as they are all together.  Which testing code are you referring to?

Reply:
When we do integration tests, sometimes we control some parameters in the tests to reduce the test running time. For example, https://github.com/microsoft/nlp-recipes/blob/master/tests/integration/test_notebooks_entailment.py. This is done by setting the cell with all the constants a "parameter" cell and pass the parameter values from the test.

Reply:
I personally prefer grouping them for the reasons @hlums mentioned. You can add a comment before every group of params (for ex *model params*, *dataset params*, etc...) to make it clearer. Anyway, we don't have strong requirements and feel free to keep it that way.

Reply:
grouped contants together and added comments before each group